content/lectures/lec02.md

```diff
@@ -6,6 +6,80 @@ assignments: ["hw0"]
 draft: false
 ---
 
+# Lecture Two: Docker
+
+## Portability
+
+Recall all of the issues that we've run into trying transfer code that we developed on our local machine to a remote host in the cloud. We want to be able to take our application code, and seamlessly run it on another machine. Along with Python dependencies, we needed to make sure that system resources, including Python and Poetry themselves, were also installed as expected. How can we avoid all of the dependency and versioning issues we ran into when we manually deployed to EC2?
+
+We were able to use Poetry and get all of the *reproducibility* benefits when we deployed to EC2, but just because a build is reproducible does *not* mean that it's **portable**.
+
+The intuition here is that we essentially want to _send a copy of our local computer_ to whoever is going to be running our code.
```

Comment:
I don't think this line translates to paper because it's very imprecise. The intuition, in a more precise manner, is that we want to send the other person a machine will all the dependencies pre-installed